### ISSUE TITLE
Upgrades the type for nptyping wrongly

### ISSUE THREAD

**Issue by mosauter** [asottile/pyupgrade#654]
I have the following example:

```py
import numpy as np
from nptyping import Float, NDArray, Shape

arr: NDArray[Shape["4 points, [x, y]"], Float] = np.array(
    [[1, 2], [2, 3], [2, 3], [2, 3]]
)

print(arr)
```

When I execute `pyupgrade` on this (with only `--py38-plus`, but the same issue is also there with `--py311-plus`), nothing happens as expected. Now I want to upgrade to new and shiny typedefinitions from python (i.e. `str | None` instead of `Optional[str]`) and import `from __future__ import annotations`.

When reexecuted `pyupgrade` correctly updates all the other code in the file as expected. But it breaks the code above. The actual resulting file is:

```py
from __future__ import annotations

import numpy as np
from nptyping import Float, NDArray, Shape

arr: NDArray[Shape[4 points, [x, y]], Float] = np.array( # notice missing '"' around parameter of Shape
    [[1, 2], [2, 3], [2, 3], [2, 3]]
)

print(arr)
```

If you try to execute that you will get the following error:
```
File "test.py", line 6
    arr: NDArray[Shape[4 points, [x, y]], Float] = np.array(
                         ^
SyntaxError: invalid syntax
```

So in short, this only happens with future-annotations and I would guess that this is a bug especially as it happens across different `pyXX-plus`.

Versions used:
- nptyping==2.1.1
- numpy==1.21.4
- pyupgrade==2.34.0

**Comment by asottile** [comment]
nptyping violates PEP 484 and is not supported

https://peps.python.org/pep-0484/#forward-references

> The string literal should contain a valid Python expression (i.e., compile(lit, '', 'eval') should be a valid code object) and it should evaluate without errors once the module has been fully loaded.
